### PR TITLE
Fix android hold action

### LINKED ISSUE #1950: Lovelace long-press does not work on Android/Chrome
**Home Assistant release with the issue:**
0.81.1

**Last working Home Assistant release (if known):**
n/a

**Browser and Operating System:**
Android/Chrome

**Description of problem:**
Testing out the long-press support from #1848...

I have a light configured with tap_action: toggle and hold_action: more-info
Long-press works fine on chrome desktop, but not in android.
When in android I have to be very careful to release the press at the same time as the "transparent circle" is drawn... 
If i keep pressing too long, it will not register the long-press and not show the more-info dialog.

### PR DESCRIPTION
**Pull request by thomasloven**
This could possibly fix #1950.

I don't have an android device with chrome, so I can't test, but this prevents the context menu from popping up when emulating a touch device in chrome on my computer - which I think was the problem.

It would be great if someone could test this before merging.
@zsarnett ?

### CODE REVIEW

**Comment by zsarnett** [comment]
Will test as soon as I can

**Comment by thomasloven** [comment]
Well... back to the drawing board...

**Comment by thomasloven** [comment]
Could someone with an android device could go [here](https://codepen.io/thomasloven/full/xQVwZV), click and long-press each of the squares and then tell me what happens?

**Comment by zsarnett** [comment]
REsults

Holding ^

Tap

**Comment by zsarnett** [comment]
I don't know enough about this code and touch events to be able to review the code to say its "good" for merging. But It does work. Requesting @balloob to review

**Comment by zsarnett** [comment]
Tested: Hold and Tap Action work. Scrolling also works

**Comment by thomasloven** [comment]
The problem, for reference, was that chrome for android opens a context menu when pressing and holding for about half a second. When the context menu is opened, the touch event is interrupted, so no `touchend` is sent. Instead a `touchcancel` is sent, and for some reason this happens even if there is no context menu...

File: src/panels/lovelace/common/directives/long-press-directive.ts [hunk 1]
```diff
@@ -64,34 +65,59 @@ class LongPress extends HTMLElement implements LongPress {
...
+      this.timer = window.setTimeout(() => {
+        this.startAnimation(x, y);
+        this.held = true;
+      }, this.holdTime);
+    };
+
+    const clickEnd = () => {
+      clearTimeout(this.timer);
       this.stopAnimation();
+      if (isTouch && this.timer == null) {
```
Comment (balloob) [inline src/panels/lovelace/common/directives/long-press-directive.ts]: you're checking if `timer == null` but `timer` is never set to `null`, just `undefined`. 

With just `==`, a comparison with `null` also matches `undefined`, but if we really only expect `undefined`, we should check for that.

image

**Comment by casey-hill** [comment]
I'm not sure if this thread is still going on but the problem is interesting.  I was able to use this thread to solve my own problem.  While the fix is in place for the phones, a Windows 10 devices running Edge or Chrome isn't working using the toggle action.  No idea how to troubleshoot something like this but my phone works fine … yet the computer(s) do not.

**Comment by thomasloven** [comment]
Avoid discussion in closed PRs.
If it's still not working for you with the lastest version of Home Assistant, please open a new issue.

**Pull request merged**